src/com/hms/flexyosisoftconnector/OSIsoftServer.java

```diff
@@ -243,6 +273,126 @@ public static void endBatch()
       batchBuffer.append("}");
    }
 
+   // omf versions of above functions
+
+   /**
+    * call this first when constructing OMF message
+    */
+   public static void startOMFdataMessage()
+   {
+      batchCount = 0;
+      batchBuffer.clearString();
+      batchBuffer.append("[");
+   }
+
+
+   /**
+    * call this for each new data point to omf message
+    */
+   public static void addPointToOMFdataMessage(String tagValue, String timestamp)
+   {
+      batchCount++;
+
+
+      batchBuffer.append("{");
+      batchBuffer.append(" \"timestamp\": \""+timestamp+".000Z\","); // should move the timezone part into the timestamp string
+      batchBuffer.append(" \"tagValue\": \""+ tagValue+"\"");
+      batchBuffer.append("}");
+
+   }
+
+   /**
+    * call when adding new tag to omf message to reference container id
+    * @param tagName
+    */
+    public static void addContainerStartToOMFdataMessage(String tagName)
+    {
+        batchBuffer.append("{");
+        // each tag has a container id that is set to the tag name
+        batchBuffer.append("\"containerid\": \""+tagName+"\"");
+        batchBuffer.append(",");
+        batchBuffer.append("\"values\": [");
+
+   }
+
+   /**
+    * call after finished adding all data points to close values array in json
+    */
+   public static void addContainerEndToOMFdataMessage()
+   {
+       batchBuffer.append("]");
+       batchBuffer.append("}"); // closes off the end of the container specific portion of the data message
+   }
+
+   /**
+    *  call this before each call to addPointToOMFdataMessage but not before the first one
+    */
+   public static void seperateDataMessage()
+   {
+      batchBuffer.append(",");
+   }
+
+   /**
+    * call this first when finished building up the OMF message
+    */
+   public static void endOMFdataMessage()
+   {
+      batchBuffer.append("]");
+   }
+
+    /**
+     * should be called for every tag in config list at constructor
+     */
+    public static void setContainerJson(ArrayList tagList) {
+       startOMFdataMessage();
+
+       for (int i = 0; i < tagList.size(); i++) {
+
+          String tagName = ((Tag) tagList.get(i)).getTagName();
+
+          // after first tag, seperate by comma
+          if(i > 0 ) {
+             seperateDataMessage();
+          }
+
+          batchBuffer.append("{");
+          batchBuffer.append("\"id\": \""+tagName+"\",");
+          batchBuffer.append("\"typeid\": \""+typeID+"\"");
+          batchBuffer.append("}");
+       }
+
+       endOMFdataMessage();
+    }
+
+   public static boolean postOMFBatch()
+   {
+
+      String url =  targetURL +"omf";
+      postHeaders = "Authorization=Basic " + authCredentials + "&Content-Type=application/json&X-Requested-With=JSONHttpRequest&messageformat=json&omfversion=1.1";
+      String postHeaderType = "&messagetype=type";

```

Comment:
This already appeared in this file already, shouldnt it just be set once?

Reply:
this is a mistake, will fix